pkg/api/addons_test.go

```diff
@@ -784,9 +787,2002 @@ func TestSetAddonsConfig(t *testing.T) {
+							CPURequests:    "88m",
+							MemoryRequests: "204Mi",
+							CPULimits:      "88m",
+							MemoryLimits:   "204Mi",
+						},
+						{
+							Name:           "heapster-nanny",
+							Image:          specConfig.KubernetesImageBase + K8sComponentsByVersionMap["1.12.8"]["addonresizer"],
+							CPURequests:    "88m",
+							MemoryRequests: "204Mi",
+							CPULimits:      "88m",
+							MemoryLimits:   "204Mi",
+						},
+					},
+				},
+				{
+					Name:    TillerAddonName,
+					Enabled: to.BoolPtr(DefaultTillerAddonEnabled),
+				},
+				{
+					Name:    ACIConnectorAddonName,
+					Enabled: to.BoolPtr(false),
+				},
+				{
+					Name:    ClusterAutoscalerAddonName,
+					Enabled: to.BoolPtr(true),
+					Config: map[string]string{
+						"scan-interval":                         "1m",
+						"expendable-pods-priority-cutoff":       "-10",
+						"ignore-daemonsets-utilization":         "false",
+						"ignore-mirror-pods-utilization":        "false",
+						"max-autoprovisioned-node-group-count":  "15",
+						"max-empty-bulk-delete":                 "10",
+						"max-failing-time":                      "15m0s",
+						"max-graceful-termination-sec":          "600",
+						"max-inactivity":                        "10m0s",
+						"max-node-provision-time":               "15m0s",
+						"max-nodes-total":                       "0",
+						"max-total-unready-percentage":          "45",
+						"memory-total":                          "0:6400000",
+						"min-replica-count":                     "0",
+						"namespace":                             "kube-system",
+						"new-pod-scale-up-delay":                "0s",
+						"node-autoprovisioning-enabled":         "false",
+						"ok-total-unready-count":                "3",
+						"scale-down-candidates-pool-min-count":  "50",
+						"scale-down-candidates-pool-ratio":      "0.1",
+						"scale-down-delay-after-add":            "10m0s",
+						"scale-down-delay-after-delete":         "1m",
+						"scale-down-delay-after-failure":        "3m0s",
+						"scale-down-enabled":                    "true",
+						"scale-down-non-empty-candidates-count": "30",
+						"scale-down-unneeded-time":              "10m0s",
+						"scale-down-unready-time":               "20m0s",
+						"scale-down-utilization-threshold":      "0.5",
+						"skip-nodes-with-local-storage":         "false",
+						"skip-nodes-with-system-pods":           "true",
+						"stderrthreshold":                       "2",
+						"unremovable-node-recheck-timeout":      "5m0s",
+						"v":                                     "3",
+						"write-status-configmap":                "true",
+						"balance-similar-node-groups":           "true",
+					},
+					Pools: []AddonNodePoolsConfig{
+						{
+							Name: "pool1",
+							Config: map[string]string{
+								"min-nodes": "5",
+								"max-nodes": "100",
+							},
+						},
+						{
+							Name: "pool2",
+							Config: map[string]string{
```

Comment:
why is backcompat making the min/max node config for each pool different? the way I read this, the used-to-be cluster-wide min/max config is now per pool. It would make sense to apply the old cluster-wide config to every node pool.

Reply:
It's the conservative approach: we assign to both the min and max the number of nodes in the pool, which essentially enables cluster-autoscaler against the nodepool, but with a configuration that won't do anything.

I'll strengthen the docs around this.